### PR TITLE
Add new source www.wuqimh.com

### PR DESCRIPTION
**Pull request by Trim21**

### CODE REVIEW

File: src/zh/wuqimanga/src/eu/kanade/tachiyomi/extension/zh/wuqimanga/WuqiManga.kt [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    private class GenreFilter(genres: Array<String>) : Filter.Select<String>("Genre", genres)
+
+
+    override fun getFilterList() = FilterList(
+        GenreFilter(getGenreList())
+    )
+
+    private fun getGenreList() = arrayOf(
+        "All"
+    )
```
Comment (arkon) [inline src/zh/wuqimanga/src/eu/kanade/tachiyomi/extension/zh/wuqimanga/WuqiManga.kt]: Are you planning on updating this? Doesn't seem very useful.

File: src/zh/wuqimanga/src/eu/kanade/tachiyomi/extension/zh/wuqimanga/WuqiManga.kt [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+class WuqiManga : ParsedHttpSource() {
+
+    override val name = "57漫画"
+    override val baseUrl = "http://www.wuqimh.com"
+    override val lang = "zh"
+    override val supportsLatest = false
+    val imageServer = arrayOf("http://images.lancaier.com")
+
+    private fun log(msg: String) = Log.i("tachiyomi", msg)
```
Comment (arkon) [inline src/zh/wuqimanga/src/eu/kanade/tachiyomi/extension/zh/wuqimanga/WuqiManga.kt]: Remove this and its usages.

File: src/zh/wuqimanga/src/eu/kanade/tachiyomi/extension/zh/wuqimanga/WuqiManga.kt [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+import org.jsoup.nodes.Element
+import com.squareup.duktape.Duktape
+
+class WuqiManga : ParsedHttpSource() {
+
+    override val name = "57漫画"
+    override val baseUrl = "http://www.wuqimh.com"
+    override val lang = "zh"
+    override val supportsLatest = false
+    val imageServer = arrayOf("http://images.lancaier.com")
```
Comment (arkon) [inline src/zh/wuqimanga/src/eu/kanade/tachiyomi/extension/zh/wuqimanga/WuqiManga.kt]: This doesn't need to be an array if there's only one thing.

File: src/zh/wuqimanga/src/eu/kanade/tachiyomi/extension/zh/wuqimanga/WuqiManga.kt [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        return GET("$baseUrl/search/q_$query-p-$page", headers)
+    }
+
+    override fun searchMangaNextPageSelector() = "div.book-result > div > span > a.prev"
+    override fun searchMangaSelector() = "div.book-result li.cf"
+    override fun searchMangaFromElement(element: Element): SManga {
+        Log.i("57M", "try to parse el")
```
Comment (arkon) [inline src/zh/wuqimanga/src/eu/kanade/tachiyomi/extension/zh/wuqimanga/WuqiManga.kt]: Remove this.

File: src/zh/wuqimanga/src/eu/kanade/tachiyomi/extension/zh/wuqimanga/WuqiManga.kt [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+                }
+            }
+        }
+        manga.thumbnail_url = element.select("a.bcover > img").attr("src")
+        return manga
+    }
+
+    override fun mangaDetailsRequest(manga: SManga) = GET("$baseUrl/${manga.url}", headers)
+    override fun chapterListRequest(manga: SManga) = mangaDetailsRequest(manga)
+    //    override fun chapterListSelector(): String = "#non-exist-id-23123"
```
Comment (arkon) [inline src/zh/wuqimanga/src/eu/kanade/tachiyomi/extension/zh/wuqimanga/WuqiManga.kt]: Remove this.

**Comment by as280093** [comment]
**Update Icons to This:**
[ic_launcher .zip](https://github.com/inorichi/tachiyomi-extensions/files/4279338/ic_launcher.12.zip)

File: src/zh/wuqimanga/src/eu/kanade/tachiyomi/extension/zh/wuqimanga/WuqiManga.kt (hunk 1 above)
Comment (Trim21) [reply src/zh/wuqimanga/src/eu/kanade/tachiyomi/extension/zh/wuqimanga/WuqiManga.kt]: Source doesn't support filter

File: src/zh/wuqimanga/src/eu/kanade/tachiyomi/extension/zh/wuqimanga/WuqiManga.kt [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+    private class GenreFilter(genres: Array<String>) : Filter.Select<String>("Genre", genres)
+
+
+    override fun getFilterList() = FilterList(
+        GenreFilter(getGenreList())
+    )
+
+    private fun getGenreList() = arrayOf(
+        "All"
+    )
```
Comment (arkon) [inline src/zh/wuqimanga/src/eu/kanade/tachiyomi/extension/zh/wuqimanga/WuqiManga.kt]: Can just remove the filter code then.

**Comment by arkon** [comment]
Can you revert the change to `BainianManga.kt`?

**Comment by Trim21** [comment]
@arkon awkward....done

**Pull request merged**